bears/installation_tool.py

```diff
@@ -0,0 +1,158 @@
+                                requirements failed installing.
+    :return:                    A list with the packages which had their
+                                requirements failing to be installed.
+    """
+    package_object = importlib.import_module(
+        package_name + "." + package_name)
+    for requirement in getattr(package_object, package_name).REQUIREMENTS:
+        try:
+            subprocess.call([requirement.install_command()])
+        except:
+            package_failed_list.append(package_name)
+    return package_failed_list
+
+
+def get_all_bears_names_from_PyPI():
+    """
+    Gets all the bears names from PyPI, using the link in the description.
+
+    :return: A list with all the bear names.
+    """
+    output = get_output(['pip', 'search', "coala.rtfd.org"])
+    return re.findall(r'(.+) \(\d.*\n', output)
+
+
+def install_bears(bear_names_list, arg):
+    """
+    Tries to install each bear from the bear_names_list. Will also check for
+    bears which failed to be installed, or their requirements failed to be
+    installed.
+
+    :param bear_names_list: The list which contains the names of the bears.
+    :param arg:             An arg which is given to also install the bears'
+                            dependencies.
+    """
+    for bear_name in bear_names_list:
+        bear_name = bear_name.replace(' ', '')
+        bears_failed_list = []
+        try:
+            install_pip_package(bear_name)
+        except:
+            bears_failed_list.append(bear_name)
+
+        if arg:
+            bears_failed_list = install_requirements(
+                bear_name, bears_failed_list)
+
+        check_failed_bears(bears_failed_list)
+
+
+def create_installation_parser():
+    """
+    Creates a parser for command line arguments.
+
+    :return: Parser arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description='Install bears requirements.')
+    parser.add_argument('-i', '--install',
+                        help='Install the requirements',
+                        action='store_true')
+    return parser
+
+
+def main():
+    args = create_installation_parser().parse_args()
+
+    option_completer = WordCompleter(['All', 'Some', 'None'], ignore_case=True)
+
+    # HACK using SqlLexer to color the words which are correct
+    answer = prompt('What bears would you like to install?\n'
+                    'All / Some / None\n',
+                    lexer=SqlLexer, completer=option_completer)
+
+    if answer in ("All", "all"):
+        print('Great idea, we are installing all the bears right now.')
+
+        install_bears(sorted(get_all_bears_names_from_PyPI()), args.install)
+
+    elif answer in ("Some", "some"):
+        print('This is a list of all the bears you can install:')
+
+        bear_name_completer = WordCompleter(
+            get_all_bears_names_from_PyPI(), ignore_case=True)
+
+        for bear_name in sorted(get_all_bears_names_from_PyPI()):
```

Comment:
You're doing `get_all_bears_names_from_PyPI` twice, which will take a lot of time since it's a network request. Maybe store it once and reuse it everywhere?